compatibility-verifier/compCheck.sh

```diff
@@ -0,0 +1,210 @@
+    exit 1
+  fi
+else
+  # use a temp directory in case workingDir is not provided
+  tmpDir=$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')
+  workingDir=$tmpDir
+fi
+
+
+# This function builds Pinot given a specific commit hash and target directory
+function checkoutAndBuild() {
+  if [ $# -ne 2 ]; then 
+    usage checkoutAndBuild
+  fi
+  commitHash=$1
+  targetDir=$2
+  
+  pushd "$targetDir" || exit 1
+  git init
+  git remote add origin https://github.com/apache/incubator-pinot
+  git fetch --depth 1 origin "$commitHash"
+  git checkout FETCH_HEAD
+  mvn install package -DskipTests -Pbin-dist
+  popd || exit 1
+}
+
+# Given a component and directory, start that version of the specific component 
+function startService() {
+  if [ $# -ne 2 ]; then 
+    usage startService
+  fi
+  serviceName=$1
+  dirName=$2
+  if [[ ! " ${components[*]} " =~ $serviceName ]]; then
+    printf "Not a valid component. Needs to be one of: %s \n" "${components[*]}"
+    exit 1
+  fi
+  # Upon start, save the pid of the process for a component into a file in /tmp/{component}.pid, which is then used to stop it
+  pushd "$dirName"/pinot-tools/target/pinot-tools-pkg/bin  || exit 1
+  if [ "$serviceName" = "zookeeper" ]; then
+    sh -c 'echo $$ > /tmp/zookeeper.pid; exec ./pinot-admin.sh StartZookeeper' &
+  elif [ "$serviceName" = "controller" ]; then 
+    sh -c 'echo $$ > /tmp/controller.pid; exec ./pinot-admin.sh StartController' &  
+  elif [ "$serviceName" = "broker" ]; then
+    sh -c 'echo $$ > /tmp/broker.pid; exec ./pinot-admin.sh StartBroker' &
+  elif [ "$serviceName" = "server" ]; then
+    sh -c 'echo $$ > /tmp/server.pid; exec ./pinot-admin.sh StartServer' &
+  fi 
+  popd || exit 1
+}
+
+# Given a component, check if it known to be running and stop that specific component
+function stopService() {
+  if [ $# -ne 1 ]; then 
+    usage stopService
+  fi
+  serviceName=$1
+  if [[ ! " ${components[*]} " =~ $serviceName ]]; then
+    printf "Not a valid component. Needs to be one of: %s \n" "${components[*]}"
+    exit 1
+  fi
+  if [ -f /tmp/"${serviceName}".pid ]; then 
+    servicePid=$(</tmp/"${serviceName}".pid)
+    rm /tmp/"${serviceName}".pid
+    if [ -n "$servicePid" ]; then
+      kill -9 "$servicePid"
+    fi
+  else
+    echo "Can't stop component ${serviceName} since can't component ${serviceName} is not known to have been started."
+  fi
+}
+
+# Starts a Pinot cluster given a specific target directory
+function startServices() {
+  dirName=$1
+  startService zookeeper "$dirName"
+  startService controller "$dirName"
+  startService broker "$dirName"
+  startService server "$dirName"
+  echo "Cluster started."
+}
+
+# Stops the currently running Pinot cluster
+function stopServices() {
+  stopService controller
+  stopService broker
+  stopService server
+  stopService zookeeper
+  echo "Cluster stopped."
+} 
+
+# upgrades the cluster by upgrading components one by one (Controller -> Broker -> Server) 
```

Comment:
We need to run tests before/after each step, so there will be a lot of code between each stop and start. Don't put this as a function as yet.. It can be in the main code for now.